docs/content/doc/installation/from-source.en-us.md

```diff
@@ -15,68 +15,95 @@ menu:
 
 # Installation from source
 
-This section will not include basic [installation instructions](https://golang.org/doc/install).
+You should [install go](https://golang.org/doc/install) and set up your go
+environment correctly. In particular, it is recommended to set the `$GOPATH`
+environment variable and to add the go bin directory or directories
+`${GOPATH//://bin:}/bin` to the `$PATH`. See the Go wiki entry for
+[GOPATH](https://github.com/golang/go/wiki/GOPATH).
 
-**Note**: Go version 1.8 or higher is required
+**Note**: Whilst building, Gitea will download and build several packages
+necessary to build itself. If you don't add the go bin directory to executable
+path you will have to manage this yourself.
+
+**Note 2**: Go version 1.8 or higher is required, however it is recommended to
+obtain the same version as our continuous integration, see the advice given in
+<a href='{{ relref "docs/advanced/hacking-on-gitea.en-us.md" }}'>Hacking on
+Gitea</a>
 
 ## Download
 
-First retrieve the source code. The easiest way is to use the Go tool. Use the following
-commands to fetch the source and switch into the source directory.
+First retrieve the source code. The easiest way is to use the Go tool. Use the
+following commands to fetch the source and switch into the source directory.
+Go is quite opinionated about where it expects its source code, and simply
+cloning the Gitea repository to an arbitrary path is likely to lead to
+problems - the fixing of which is out of scope for this document.
 
-```
+```bash
 go get -d -u code.gitea.io/gitea
-cd $GOPATH/src/code.gitea.io/gitea
+cd "$GOPATH"/src/code.gitea.io/gitea
```

Comment:
I think ending quote should be at the and of line

Reply:
It doesn't really matter AFAIU, but I think keeping the quote next to variable keeps it with its context. 

If we had `"$GOPATH/src/code gitea io/gitea"` then we would have to extend the quoted section, but as it stands the quote just needs to be around bit that could possibly have a space. I would probably prefer that to be `"$GOPATH"/src/"code gite io"/gitea` though.

If you much prefer extending the quote then I will change.

Reply:
is the quote even needed?

Reply:
Yeah if $GOPATH contains a space. It's good practice to put environment variables in quotes in general.

Reply:
I think all path must be quoted if any part of it contains space "$GOPATH/src/code gite io/gitea"

Reply:
```bash
export TD="A Test Directory"
mkdir "$TD"
echo "foo" > "$TD"/foo
mkdir -p "$TD"/no-space/"Some Spaces"
```

Reply:
personally have not used such syntax... putting quotes around full path seems more natural to me but that is just my personal preference

Reply:
OK I've extended the quotes.

Reply:
thanks :)